clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ConsumerAction.java

```diff
@@ -27,6 +27,8 @@ public class ConsumerAction implements Comparable<ConsumerAction> {
   private boolean isHelixTriggeredAction = true;
   private CompletableFuture<Void> future = new CompletableFuture<>();
 
+  private TopicPartitionReplicaRole.VersionRole versionRole;
+
   public ConsumerAction(
```

Comment:
Maybe in the near future, we can refactor this `ConsumerAction` class as it becomes a little messy now since some of the params are not useful or required in certain actions. For example, this new param: `versionRole` is only useful for some of the actions, such as `kill` might not need this param.
During the refactoring, we can consider to use `inheritance` to specify different behaviors for different `ConsumerAction` type.